enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,283 @@ any user-defined primary network.
+| IPAMLifecycle  | Control IP addresses management lifecycle. When `Persistent` is specified it enable workloads have persistent IP addresses. For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots). Supported by Topology `Layer2` & `Localnet`.                                                                                                         | Yes      |
+ 
+The cluster scoped CRD should have the following additional field:
+
+| Field name        | Description                                                                                                   | optional |
+|-------------------|---------------------------------------------------------------------------------------------------------------|----------|
+| NamespaceSelector | List of the standard `metav1.LabelSelector` selector for which namespace the network should be available for. | No       |
+| Template          | The user defined network spec.                                                                                | No       |
+
+The template type should be the namespace scope CRD spec.
+
+> **Note:** The spec should be extended with care and strive to have minimal set of fields to provide nice abstraction for the NAD spec.
+
+##### Status
+The CRD status should reflect the NAD state through conditions.
+For example, when the NAD its created the condition should be placed with status `True`.
+
+For cluster scoped networks, the condition should be true once all desired namespaces are provisioned with the corresponding NAD.
+
+The cluster scoped CRD status should reflect on which namespaces the network is available.
+
+##### CRD Definitions
+
+The following CRDs should be installed by [CNO](https://github.com/openshift/cluster-network-operator) and managed by OVN-Kubernetes.
+
+##### Namespace scoped CRD
+```golang
+// UserDefinedNetwork describe network request for a Namespace
+type UserDefinedNetwork struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Spec is immutable"
+    // +kubebuilder:validation:XValidation:rule=" self.topology != 'Layer3' || (self.topology == 'Layer3' && size(self.subnets) > 0)", message="Subnets is required for Layer3 topology"
+    // +kubebuilder:validation:XValidation:rule="self.topology != 'Localnet' || (self.topology == 'Localnet' && self.role == 'Secondary')", message="Topology localnet is not supported for primary network"
```

Comment:
Localnet should be LocalNet to be PascalCase which is the expected format for enums

Reply:
Hello @JoelSpeed, sorry for opening this old thread, but localnet is a single word, defined in https://www.ovn.org/support/dist-docs/ovn-architecture.7.html. The origin of it may be local network, but in our case, we simply refer to the feature of OVN which is called "localnet". This name is well established and AFAIK was never spelled as two different words. I suggest we call this value `Localnet` to avoid confusion.

Reply:
Has this API shipped? Is it in the o/api repo? 

You're correct, given the usage of this as a keyword in the OVN architecture, we should change this to `Localnet`, but, we need to understand what the implications of changing this are.

If the feature is still in techpreview, we can change this before we go to GA